src/Microsoft.AspNetCore.Razor.Evolution/CodeGeneration/DesignTimeCSharpRenderer.cs

```diff
@@ -77,7 +77,13 @@ public override void VisitUsingStatement(UsingStatementIRNode node)
 
         public override void VisitCSharpStatement(CSharpStatementIRNode node)
         {
-            if (node.Source != null)
+            if (node.Source != null && string.IsNullOrWhiteSpace(node.Content))
```

Comment:
Super solid point. With how the code is currently you have the same problem though.